src/games/strategy/engine/framework/mapDownload/DownloadRunnable.java

```diff
@@ -52,26 +59,45 @@ public void setError(final String error) {
 
   @Override
   public void run() {
+    if (urlString.startsWith("http")) {
+      downloadFile();
+    } else {
+      readLocalFile();
+    }
+  }
+
+  private void downloadFile() {
     URL url;
+
     try {
-      // System.out.println(System.getProperty("http.proxyHost"));
-      // System.out.println(System.getProperty("http.proxyPort"));
       url = new URL(urlString.trim());
     } catch (final MalformedURLException e1) {
       error = "invalid url";
       return;
     }
+    try {
+      HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+
+      int status = conn.getResponseCode();
+      if (status == HttpURLConnection.HTTP_MOVED_TEMP || status == HttpURLConnection.HTTP_MOVED_PERM
+          || status == HttpURLConnection.HTTP_SEE_OTHER) {
+
+        String newUrl = conn.getHeaderField("Location");
+        conn = (HttpURLConnection) new URL(newUrl).openConnection();
+      }
 
-    try (InputStream stream = url.openStream();
-        ByteArrayOutputStream sink = new ByteArrayOutputStream()) {
-      InstallMapDialog.copy(sink, stream);
-      contents = sink.toByteArray();
+      try (InputStream stream = conn.getInputStream()) {
+        final ByteArrayOutputStream sink = new ByteArrayOutputStream();
+        InstallMapDialog.copy(sink, stream);
+        contents = sink.toByteArray();
+      }
     } catch (final Exception e) {
       error = e.getMessage();
+      return;
```

Comment:
Adding this return  is not very ideal.. but it's helpful in the meantime for the reader to have this signal that there is early termination here.